### PR TITLE
smalls improvement to GL backend stencil and raster state

### PR DESCRIPTION
**Pull request by pixelflinger**
reviewers: it'll much easier to review each CL separately. thanks.

### CODE REVIEW

File: filament/backend/include/backend/DriverEnums.h [hunk 1]
```diff
@@ -1002,29 +979,37 @@ struct StencilState {
...
             //! Masks the bits of the stencil values participating in the stencil comparison test.
-            uint8_t readMask                                : 8;                    // 24
+            uint8_t readMask;                                                       // 24
 
             //! Masks the bits of the stencil values updated by the stencil test.
-            uint8_t writeMask                               : 8;                    // 32
+            uint8_t writeMask;                                                      // 32
         };
         uint32_t u = 0;
     };
```
Comment (bejado) [inline filament/backend/include/backend/DriverEnums.h:988]: Do we even need this `union` anymore?

Comment (pixelflinger) [reply filament/backend/include/backend/DriverEnums.h:988]: that's a good point, let me play with it. probably not.

File: filament/backend/include/backend/DriverEnums.h [hunk 2]
```diff
@@ -960,29 +960,6 @@ struct PolygonOffset {
 struct StencilState {
     using StencilFunction = SamplerCompareFunc;
 
-    StencilState() noexcept : referenceValue(0u), stencilWrite(false), padding(0u) {
-        static_assert(sizeof(StencilOperations) == sizeof(uint32_t), "StencilOperations size not what was intended");
-        static_assert(sizeof(StencilState) == 12u, "StencilState size not what was intended");
```
Comment (bejado) [inline filament/backend/include/backend/DriverEnums.h:965]: As long as we have padding bytes, it's probably a good idea to keep this assertions.

Comment (bejado) [reply filament/backend/include/backend/DriverEnums.h:965]: Oh just kidding, I see you put them below 😄

File: filament/backend/include/backend/DriverEnums.h (hunk 1 above)
Comment (bejado) [reply filament/backend/include/backend/DriverEnums.h:988]: Note that this struct may grow—we can actually move the stencil reference value here, as it's also set separately per front/back facing polygons.

File: filament/backend/include/backend/DriverEnums.h (hunk 2 above)
Comment (pixelflinger) [reply filament/backend/include/backend/DriverEnums.h:965]: I just moved it.

**Review by bejado — APPROVED** [review]

**Pull request merged**